ta/crypto_perf/ta_crypto_perf.c

```diff
@@ -465,8 +467,475 @@ TEE_Result cmd_hash_prepare_op(uint32_t param_types, TEE_Param params[4])
 	return TEE_SUCCESS;
 }
 
+struct attr_packed {
+	uint32_t id;
+	uint32_t a;
+	uint32_t b;
+};
+
+static TEE_Result unpack_attrs(const uint8_t *buf, size_t blen,
```

Comment:
I'm worried about this function being called on each outer loop. The time stats will include the time it takes to unpack all attributes. Is that something we want in our stats? IMO, the outer loop should only be there to include the context switch cost, not extra operations like this one.

Reply:
I find this packing and unpacking is a part of CA-TA of hpre.I do not have any good way to solve it.Do you have any suggestions?

Reply:
The operation exists in every TA function of asym.We calculate time at CA side.It is impossible to do this in prepare step.

Reply:
> The operation of unpack_attrs exists in every TA function of asym.

That's precisely the issue with `unpack_attrs()` being included in the benchmark loop.

> We calculate time at CA side.It is impossible to do this in prepare step.

How is that impossible?

Reply:
If we do unpacking at CA side, it will be difficult to transport attrs from ca to ta.If we do unpacking at TA side, it will be difficult to remove unpacking time. Or we should use global variable to save atrrs before that?

Reply:
Still not addressed

Reply:
Explain how that is solved. Remove the "solved" marker.

Reply:
It looks to me that `unpack_attrs()` is called from `cmd_asym_prepare_attrs()`, `cmd_asym_prepare_keypair()` and `cmd_asym_prepare_enc_sign()` so not in the critical sequence for performance measurements. Am I right?

Reply:
My bad, `cmd_asym_prepare_enc_sign()`/`TA_CRYPTO_PERF_CMD_ASYM_CIPHER_PREPARE_ENC_SIGN` should be renamed `cmd_asym_encrypt_sign`/`TA_CRYPTO_PERF_CMD_ASYM_ENCRYPT_SIGN`, it's not a a preparation step.

Reply:
My-rebad: `prepare_enc_sign()` in _host/xtest/asym_perf.c_ is out of the perf monitoring loop, so it seems to me the issue is addressed.